### PR TITLE
Implement TeV catalog

### PR DESCRIPTION
**Pull request by vorugantia**
In this PR:
- Add TeV to ng2-select (and fix ng2-select for the other catalogs)
- Add TeV catalog to map

### CODE REVIEW

File: src/app/widgets/map/map.component.ts [hunk 1]
```diff
@@ -61,39 +74,40 @@ export class MapComponent implements OnInit, OnDestroy {
     this.map.addCatalog(cat);
 
 
-    var n_sources = Object.keys(catalog.Source_Name).length;
+    var n_sources = catalog.data.index.length;
```
Comment (cdeil) [inline src/app/widgets/map/map.component.ts:77]: Define `catalog.length` that re-exposes this?

File: src/app/widgets/map/map.component.ts [hunk 2]
```diff
@@ -61,39 +74,40 @@ export class MapComponent implements OnInit, OnDestroy {
...
+      // var row = i.toString();
 
-      var round_ra = (Math.round(catalog['RAJ2000'][row] * 100) / 100).toFixed(2);
-      var round_dec = (Math.round(catalog['DEJ2000'][row] * 100) / 100).toFixed(2);
-      var round_glon = (Math.round(catalog['GLON'][row] * 100) / 100).toFixed(2);
-      var round_glat = (Math.round(catalog['GLAT'][row] * 100) / 100).toFixed(2);
+      var round_ra = (Math.round(catalog.getSourceByRowIndex(i).data[raType] * 100) / 100).toFixed(2);
+      var round_dec = (Math.round(catalog.getSourceByRowIndex(i).data[decType] * 100) / 100).toFixed(2);
+      var round_glon = (Math.round(catalog.getSourceByRowIndex(i).data.GLON * 100) / 100).toFixed(2);
+      var round_glat = (Math.round(catalog.getSourceByRowIndex(i).data.GLAT * 100) / 100).toFixed(2);
```
Comment (cdeil) [inline src/app/widgets/map/map.component.ts:87]: For this, maybe introduce a convenience method on the base catalog class that takes the column name and number of digits. Maybe call it `get_val`?

File: src/app/widgets/map/map.component.ts (hunk 1 above)
Comment (vorugantia) [inline src/app/widgets/map/map.component.ts:77]: Done

File: src/app/widgets/map/map.component.ts (hunk 2 above)
Comment (vorugantia) [inline src/app/widgets/map/map.component.ts:87]: Done

**Pull request merged**